### PR TITLE
MCR-2690 fix Solr URI resolver throws StringIndexOutOfBoundsException…

### PR DESCRIPTION
**Pull request by sebhofmann**
… if solr query is too short

[Link to jira](https://mycore.atlassian.net/browse/MCR-2690).

### CODE REVIEW

File: mycore-solr/src/main/java/org/mycore/solr/common/xml/MCRSolrQueryResolver.java [hunk 1]
```diff
@@ -42,43 +43,48 @@
...
  *
- * @author Matthias Eichner
- * @author mcrsherm
+ * @author Sebastian Hofmann
  */
 public class MCRSolrQueryResolver implements URIResolver {
 
     private static final String REQUEST_HANDLER_QUALIFIER = "requestHandler";
 
+    public static final String REQUEST_HANDLER_GROUP_NAME = "requesthandler";
```
Comment (kkrebs) [inline mycore-solr/src/main/java/org/mycore/solr/common/xml/MCRSolrQueryResolver.java:52]: Should be "requestHandler" as specified in JavaDoc

File: mycore-solr/src/main/java/org/mycore/solr/common/xml/MCRSolrQueryResolver.java [hunk 2]
```diff
@@ -42,43 +43,48 @@
...
-        } catch (IOException e) {
-            throw new TransformerException("Unable to get input stream from solr: " + solrURL.getUrl(), e);
+                try {
+                    return new MCRURLContent(solrURL.getUrl()).getSource();
+                } catch (IOException e) {
+                    throw new TransformerException("Unable to get input stream from solr: " + solrURL.getUrl(), e);
+                }
+            }
         }
+        throw new IllegalArgumentException("Didnt understand uri: " + href);
```
Comment (kkrebs) [inline mycore-solr/src/main/java/org/mycore/solr/common/xml/MCRSolrQueryResolver.java]: please fix typo Didnt -> Didn't

**Review by kkrebs — CHANGES_REQUESTED** [review]
looks good for me, only 2 little typos to fix - see code comments.

File: mycore-solr/src/main/java/org/mycore/solr/common/xml/MCRSolrQueryResolver.java (hunk 2 above)
Comment (rsteph-de) [reply mycore-solr/src/main/java/org/mycore/solr/common/xml/MCRSolrQueryResolver.java]: even better: "Did not"

**Review by kkrebs — APPROVED** [review]

**Pull request merged**